haystack/document_stores/opensearch.py

```diff
@@ -582,11 +690,12 @@ def _validate_and_adjust_document_index(self, index_name: str, headers: Optional
                 if self.knn_engine != "score_script":
                     self._validate_approximate_knn_settings(existing_embedding_field, index_settings, index_id)
 
-            # Adjust global ef_search setting (nmslib only). If not set, default is 512.
+            # Adjust global ef_search setting (nmslib only).
             if self.knn_engine == "nmslib":
                 ef_search = index_settings.get("knn.algo_param", {}).get("ef_search", 512)
-                if self.index_type == "hnsw" and ef_search != 20:
-                    body = {"knn.algo_param.ef_search": 20}
+                desired_ef_search = self.knn_parameters.get("ef_search", 20)
```

Comment:
This is the same duplicate of `ef_search` for index_type='hnsw' as mentioned before.